custom_components/samsungtv_tizen/media_player.py

```diff
@@ -959,26 +961,41 @@ async def async_select_source(self, source):
                         time.sleep(int(this_key)/1000)
                     else:
                         if this_key.startswith("ST_"):
-                            await self.hass.async_add_job(self._smartthings_keys, this_key)
+                            self.hass.async_add_job(self._smartthings_keys, this_key)
                         else:
-                            await self.hass.async_add_job(self.send_command, this_key)
+                            self.hass.async_add_job(self.send_command, this_key)
             elif source_key.startswith("ST_"):
-                await self.hass.async_add_job(self._smartthings_keys, source_key)
+                self.hass.async_add_job(self._smartthings_keys, source_key)
             else:
-                await self.hass.async_add_job(self.send_command, self._source_list[ source ])
+                self.hass.async_add_job(self.send_command, self._source_list[ source ])
         elif source in self._app_list:
             source_key = self._app_list[ source ]
-            await self.hass.async_add_job(self.send_command, source_key, "run_app")
+            self.hass.async_add_job(self.send_command, source_key, "run_app")
         elif source in self._channel_list:
             source_key = self._channel_list[ source ]
             ch_media_type = MEDIA_TYPE_CHANNEL
             if source_key.startswith("http"): ch_media_type = MEDIA_TYPE_URL
-            await self.hass.async_add_job(self.async_play_media,ch_media_type, source_key)
+            self.hass.async_add_job(self.async_play_media,ch_media_type, source_key)
         else:
             _LOGGER.error("Unsupported source")
             return
         self._last_source_time = datetime.now()
         self._source = source
+        if self._state == STATE_OFF:
+            #Try again after turning TV on if it is currently off
+            self.hass.async_add_job(self._execute_after_turn_on,"source",source)
+
+    async def _execute_after_turn_on(self, command, *arguments):
+        #Execute a function after turning on the TV first, right now only takes 'source' as command
+        _LOGGER.debug("Turning on TV first...")
+        self.turn_on(optimistic=False)
+        for _ in range(35):
```

Comment:
this must be the programming equivalent of a machine gun 😆 

i wonder if we can't move this logic to the `state()` function, polling like this makes sense for external data, but as we set the state ourselves there must be a better way of doing it..

Reply:
Hahah more like a slow shooting rifle then with this frequency, but I get your point! I'll clean it up before we merge, was a bit lazy on this one!